src/visudo/mod.rs

```diff
@@ -89,6 +91,20 @@ fn run_visudo(file_arg: Option<&str>, perms: bool, owner: bool) -> io::Result<()
     })?;
 
     let result: io::Result<()> = (|| {
+        if check {
+            let (_sudoers, errors) = Sudoers::read(&sudoers_file)?;
+
+            return if errors.is_empty() {
```

Comment:
~~I think `return`-ing from here is a good indicator that this closure should be refactored into a function for readability.~~ no longer applicable after the `check` / `run` refactor